internal/scm/gitea/gitea.go

```diff
@@ -138,15 +158,26 @@ func (g *Gitea) getRepositories(ctx context.Context) ([]*gitea.Repository, error
 	for _, repo := range allRepos {
 		repoMap[repo.ID] = repo
 	}
-	allRepos = make([]*gitea.Repository, 0, len(repoMap))
+
+	allConstructedRepos := make([]*constructedRepo, 0, len(repoMap))
```

Comment:
Topics are fetched even if no topics are provided to filter on. This will greatly increase the number of requests. Since we have no use for it other than the filtering, I would remove `constructedRepo` and pass the topics directly into the filtering when those are provided.

Reply:
I was aware of that fact and started first to implement it that way. I dislike the fact though that the behavior of the code is different based on the topic input. If you prefer this way I can change it though.

Reply:
I strongly think we should not unnecessary API calls, at least when it will increase by this much. The number of calls will go from 1 per 100 repos, to 1 per repo.